File: subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        output = []
        nth_bit = 1 << n

        for i in range(2 ** n):
            # generate bitmask, from 0..00 to 1..11
            bitmask = bin(i | nth_bit)[3:]

        # crazy hamburger
        # for i in range(2 ** n, 2 ** (n + 1)):
        #     # generate bitmask, from 0..00 to 1..11
        #     bitmask = bin(i)[3:]

            # append subset corresponding to that bitmask
            output.append([nums[j] for j in range(n) if bitmask[j] == '1'])

        return output

File: subsets/subsets.cpp
//
// Created by kolaq on 01.06.2021.
//
#include <vector>
#include <iostream>
#include <cmath>

using namespace std;

vector<vector<int>> subsets(vector<int> &nums) {
    int n = nums.size();
    int nth_bit = 1 << n;
    int bitmask = 0b0;
    vector<vector<int>> output = {{}};

    for (int i = 0; i < pow(2, n); ++i) {
        bitmask = bin()
    }

    return output;
}

int main() {
//    cout << (2 ^ 4);
}